Ukol_3/cpp/linked_list.cpp

```diff
@@ -0,0 +1,211 @@
+    }
+
+    Node* newNode = createNode(data); // vytvoreni noveho uzlu
+    Node* temp = *head;
+
+    // cyklus se opakuje dokud i neni mensi nez index-1
+    for (int i=0; temp !=nullptr && i<index-1; i++) {
+        temp = temp->next;
+    }
+
+    // pokud temporary je nebude nic
+    if (temp == nullptr) {
+        cerr<<"Index out of bounds"<<endl;
+        delete newNode;
+        return;
+    }
+
+    newNode->next = temp->next; // next u noveho uzlu bude next od temporary
+    temp->next=newNode; // next u temporary bude novy uzel
+}
+
+// Funkce pro smazání uzlu ze začátku seznamu
+void deleteAtBeginning(Node **head)
+{
+    if (*head == nullptr) return;
+    Node* temp= *head; // vytvoreni temporary uzlu
+    *head = (*head)->next; // nova head bude next od head
+    delete temp;
+     
+}
+
+// Funkce pro smazání uzlu z konce seznamu
+void deleteAtEnd(Node **head)
+{
+    if (*head==nullptr) return;
+
+    if ((*head)->next == nullptr){
+        delete *head;
+        *head = nullptr;
+        return;
+    }
+
+    Node* temp = *head;
+    while (temp->next->next != nullptr) {
+        temp = temp->next;
+    }
+
+    delete temp->next;
+    temp->next = nullptr;
+}
+
+// Funkce pro smazani uzlu na indexu
+void deleteAtIndex(Node *head, int index)
+{
+    if (head == nullptr) return;
+
+    if (index == 0) {
+        Node* temp = head;
+        head = head->next;
+        delete temp;
+        return;
+    }
+
+    Node* temp = head;
+    for (int i = 0; temp != nullptr && i < index - 1; ++i) {
+        temp = temp->next;
+    }
+
+    if (temp == nullptr || temp->next == nullptr) {
+        std::cerr << "Index out of bounds" << std::endl;
+        return;
+    }
+
+    Node* nodeToDelete = temp->next;
+    temp->next = nodeToDelete->next;
+    delete nodeToDelete;
+}
+
+// Funkce pro nalezeni prvniho vyskytu
+int findFirstOccurrence(Node *head, int value)
+{
+    int index = 0;
+    while (head!=nullptr){ // dokud head nebude nullptr (dokud nebude konec listu)
+        if (head->data==value) return index; // pokud se data od head rovna zadane hodnote, vrati se index
+        head = head->next; // head je dalsi prvek
+        index++;
+    }
+    return -1;
+}
+
+// Funkce pro třídění seznamu
+void sortList(Node **head)
+{
+    if (*head == nullptr) return;
+
+    bool swapped;
+    do {
+        swapped = false;
+        Node* temp = *head;
+        while (temp->next != nullptr) {
+            if (temp->data > temp->next->data) {
+                std::swap(temp->data, temp->next->data);
+                swapped = true;
+            }
+            temp = temp->next;
+        }
+    } while (swapped);
+}
+
+// Funkce pro smazani (dealokaci) seznamu
+void deleteList(Node **head)
+{
+    while (*head != nullptr) {
+        deleteAtBeginning(head);
+    }
+}
+
+// Operátor pro tisk dat
+std::ostream &operator<<(std::ostream &os, Node *head)
+{
+    Node* temp = head;
+    while (temp != nullptr) {
+        os << temp->data << " ";
```

Comment:
téměř vyřešeno jen vám to na výstupu stále vrací ten nullptr "1 2 3 nullptr" místo pouze dat "1 2 3"

Reply:
Lepší ale stále je tam mezera na konci řetězce navíc.